Tools/CMake/AMReXThirdPartyLibraries.cmake

```diff
@@ -57,8 +57,11 @@ endif ()
 # HYPRE
 #
 if (AMReX_HYPRE)
-    if(ENABLE_CUDA)
+    if(AMReX_CUDA)
         find_package(HYPRE 2.20.0 REQUIRED)
+        find_library(LIBCUSPARSE cusparse PATHS ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES})
+        find_library(LIBCURAND curand PATHS ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES})
+        target_link_libraries(amrex PUBLIC ${LIBCUSPARSE} ${LIBCURAND})
```

Comment:
Recommend using the CMake targets syntax

```
find_package(CUDAToolKit REQUIRED)
target_link_libraries(amrex PUBLIC 
    CUDA::cusparse    
    CUDA::curand    
    CUDA::cudart
    CUDA::cublas
    CUDA::nvToolsExt)
```

This will propagate additional include flags to application targets.

Reply:
@drangara I was just being explicit about all the additional libraries hypre links to. It is likely that `CUDA::cusparse` and `CUDA::curand` will be sufficient for the code to build as they will in turn depend on `cudart` and `cublas`. `nvToolsExt` is necessary if you use hypre profiling capabilities.